src/main/app/form/Form.ts

```diff
@@ -4,26 +4,13 @@ import { AnyObject } from '../controller/PostController';
 import { setupCheckboxParser } from './parser';
 
 export class Form {
-  constructor(private readonly form: FormContent, private formState: Partial<Case> = {}) {}
-
-  public setFormState(formState: Partial<Case>): void {
-    this.formState = formState;
-  }
-
-  public getFields(checkFields?: FormContent['fields']): FormContent['fields'] {
-    const fields = checkFields || this.form?.fields;
-    if (typeof fields === 'function') {
-      return fields(this.formState);
-    }
-
-    return fields;
-  }
+  constructor(private readonly fields: FormFields | FormFieldsFn) {}
```

Comment:
shouldn't this just be `FormFields` or can a function still be passed?

Reply:
@linusnorton I don't think a function can still be passed, but if we get rid of the FormFieldsFn here then we would have to add a <FormFields> type assertion whenever we create a new form, and I wasn't sure if that's better?

-     const form = new Form(<FormFields>this.fields);